src/transformations.jl

```diff
@@ -270,45 +266,31 @@ function fit(::Type{UnitRangeTransform}, X::AbstractMatrix{<:Real};
         Base.depwarn("fit(t, x) is deprecated: use fit(t, x, dims=2) instead", :fit)
         dims = 2
     end
-    if dims == 1
-        l, tmin, tmax = _compute_extrema(X)
-    elseif dims == 2
-        l, tmin, tmax = _compute_extrema(X')
-    else
-        throw(DomainError(dims, "fit only accept dims to be 1 or 2."))
-    end
-
-    for i = 1:l
-        @inbounds tmax[i] = 1 / (tmax[i] - tmin[i])
-    end
+    dims ∈ (1, 2) || throw(DomainError(dims, "fit only accept dims to be 1 or 2."))
+    tmin, tmax = _compute_extrema(X, dims)
+    @. tmax = 1 / (tmax - tmin)
+    l = length(tmin)
     return UnitRangeTransform(l, dims, unit, tmin, tmax)
 end
 
-function _compute_extrema(X::AbstractMatrix{<:Real})
-    n, l = size(X)
-    tmin = X[1, :]
-    tmax = X[1, :]
-    for j = 1:l
-        @inbounds for i = 2:n
-            if X[i, j] < tmin[j]
-                tmin[j] = X[i, j]
-            elseif X[i, j] > tmax[j]
-                tmax[j] = X[i, j]
-            end
-        end
+function _compute_extrema(X, dims)
```

Comment:
Better be explicit to make the code clearer:
```suggestion
function _compute_extrema(X::AbstractMatrix, dims::Integer)
```